systems/controllers/controlMexUtil.cpp

```diff
@@ -0,0 +1,335 @@
+    }
+  }
+
+  const mxArray* joint_override = myGetProperty(qp_input, "joint_pd_override");
+  int num_joint_pd_overrides = mxGetNumberOfElements(joint_override);
+  msg->num_joint_pd_overrides = num_joint_pd_overrides;
+  msg->joint_pd_override.resize(num_joint_pd_overrides);
+  for (int i=0; i < num_joint_pd_overrides; i++) {
+    msg->joint_pd_override[i].position_ind = (int32_t) mxGetScalar(myGetField(joint_override, i, "position_ind"));
+    msg->joint_pd_override[i].qi_des = mxGetScalar(myGetField(joint_override, i, "qi_des"));
+    msg->joint_pd_override[i].qdi_des = mxGetScalar(myGetField(joint_override, i, "qdi_des"));
+    msg->joint_pd_override[i].kp = mxGetScalar(myGetField(joint_override, i, "kp"));
+    msg->joint_pd_override[i].kd = mxGetScalar(myGetField(joint_override, i, "kd"));
+    msg->joint_pd_override[i].weight = mxGetScalar(myGetField(joint_override, i, "weight"));
+  }
+
+  msg->param_set_name = mxArrayToString(myGetProperty(qp_input, "param_set_name"));
+  return msg;
+}
+
+PiecewisePolynomial<double> matlabToPiecewisePolynomial(const mxArray* pobj, int index) {
+  std::vector<double> ts = matlabToStdVector<double>(mxGetField(pobj, index, "ts"));
+  const int num_segments = ts.size() - 1;
+  const mxArray *coefs = mxGetFieldSafe(pobj, index, "coefs");
+  if (mxGetNumberOfDimensions(coefs) != 3) mexErrMsgTxt("coefs should be a dimension-3 array");
+  const mwSize* dim = mxGetDimensions(coefs);
+  if (dim[0] != 6 || dim[1] != num_segments || dim[2] != 4) mexErrMsgTxt("coefs should be size 6xNx4, where N is the number of spline segments");
+
+  Map<VectorXd> coefs_flat(mxGetPrSafe(coefs), dim[0] * dim[1] * dim[2]);
+  // coefs_flat[i + j * dim[0] + k * dim[0] * dim[1]] = coefs(i, j, k);
+  std::vector<Matrix<Polynomial<double>, Dynamic, Dynamic>> poly_matrices;
+  poly_matrices.reserve(num_segments);
+  for (int j=0; j < num_segments; ++j) {
+    Matrix<Polynomial<double>, Dynamic, Dynamic> poly_matrix(6, 1);
+    for (int i=0; i < 6; ++i) {
+      poly_matrix(i) = Polynomial<double>(Vector4d(coefs_flat(i + j * 6 + 3 * 6 * num_segments),
+                                                   coefs_flat(i + j * 6 + 2 * 6 * num_segments),
+                                                   coefs_flat(i + j * 6 + 1 * 6 * num_segments),
+                                                   coefs_flat(i + j * 6 + 0 * 6 * num_segments)));
+    }
+    poly_matrices.push_back(poly_matrix);
+  }
+  return PiecewisePolynomial<double>(poly_matrices, ts);
+}
+
+void parsePositionIndices(const mxArray *pobj, std::map<std::string, VectorXi> &position_indices) {
+  int num_fields = mxGetNumberOfFields(pobj);
+  for (int i=0; i < num_fields; ++i) {
+    const mxArray* pfield = mxGetFieldByNumber(pobj, 0, i);
+    Map<VectorXd> indices_double(mxGetPrSafe(pfield), mxGetNumberOfElements(pfield));
+    VectorXi indices = indices_double.cast<int> ();
+    position_indices[std::string(mxGetFieldNameByNumber(pobj, i))] = indices;
```

Comment:
There is an error here that I only discovered last night. The fix is in #93 but we should also apply it to this branch. This line should read:

```
position_indices[std::string(mxGetFieldNameByNumber(pobj, i))] = indices.array() - 1;
```